### PR TITLE
Embed manpages into the binaries

### PR DESCRIPTION
**Pull request by haesbaert**
With this invoking one of the utilities with `-h` will display the manpage.

Since quark is not a "system utility", we can't expect people to go to the trouble of installing manpages, and typing `man ./quark-mon.8` is a pain. It's also likely that the binary is sent somewhere and thus the man relationship was lost.

I had to write a mini-utility to embed the manpage, it's a bit sad that there isn't a `mandoc -T c` to export the manpage as a C struct.

### CODE REVIEW

File: man-embedder.c [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+static int
+embed(const char *input_path, const char *ifdef_name)
+{
+	FILE	*input;
+	int	 ch, line_wrap;
+
+	if ((input = fopen(input_path, "r")) == NULL)
+		err(1, "fopen");
+
+	if (ifdef_name != NULL)
```
Comment (nicholasberlin) [inline man-embedder.c:30]: Is this ever true? If not, perhaps we should bail here instead?

File: quark-test.c [hunk 2]
```diff
@@ -16,6 +16,9 @@
 
 #include "quark.h"
 
+#define MAN_QUARK_TEST
```
[the remark this answers is not in the document]
Comment (haesbaert) [reply quark-test.c:19]: I'd rather not, but no strong feelings.

File: display_man.c [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+	pid_t	pid;
+
+	fd = mkstemp(template);
+	if (fd == -1)
+		err(1, "mkstemp");
+	if (qwrite(fd, manpage_bin, sizeof(manpage_bin)) != 0)
+		err(1, "qwrite");
+	close(fd);
+
+	if (snprintf(buf, sizeof(buf), "man %s", template) >=
```
[the remark this answers is not in the document]
Comment (haesbaert) [reply display_man.c:19]: well spot! this was from an older version where I was building the cmd line and calling system(3), my bad.

File: man-embedder.c (hunk 1 above)
Comment (haesbaert) [reply man-embedder.c:30]: I wanted it to be a bit more flexible, but yeah, not really used, I was thinking I'd end up copying this code to other places in the future where an ifdef would not be needed.

File: display_man.c (hunk 3 above)
[the remark this answers is not in the document]
Comment (haesbaert) [reply display_man.c]: zefixed

**Review by nicholasberlin — APPROVED** [review]

**Pull request merged**